Sleeping more conveniently in PostgreSQL (#71)
cybertec-postgresql/today-i-learned-content: posts/TEMPLATE/index.md

---
date: 2030-12-31 # Auto updates on publish.
title: Sleeping more conveniently in PostgreSQL
author: pavlo.golub@cybertec.at
tags: ["postgresql", "tips", "tricks", "SQL", "sleep"] # max. 10 tags; lowercase; dash-separated

> description: "How to use convinient sleep() functions inside PostgreSQL scripts" # max. 300 chars.

Reviewer: ```suggestion
description: "How to use conveniently sleep() functions inside PostgreSQL scripts" # max. 300 chars.
```

---

> The sleep functions are not overly useful and rather for demonstration purposes (creating some locking situations and then trying to find out who blocked whom) but after using the `pg_sleep()` function for years, that takes input in seconds, to my horror one day I discovered that there are more convenient functions that accept even human readable input!

Reviewer: ```suggestion
Postgres' sleep functions are not particularly useful, as they should only be utilized for demonstration purposes (like provoking locking situations and then trying to find out who blocked whom).
After years of using the `pg_sleep()` function, which takes seconds as its input argument, I one day discovered that there are more convenient functions that even accept human readable input!
```

---
date: 2030-12-31 # Auto updates on publish.
title: Sleeping more conveniently in PostgreSQL
author: pavlo.golub@cybertec.at
tags: ["postgresql", "tips", "tricks", "SQL", "sleep"] # max. 10 tags; lowercase; dash-separated

> description: "How to use conveniently sleep() functions inside PostgreSQL scripts" # max. 300 chars.

Reviewer: ```suggestion
description: "How to conveniently use sleep() functions inside PostgreSQL scripts" # max. 300 chars.
```

Reviewer: Sorry, my mistake 😅
